resources/js/components/ui/Table/TQueryTable.tsx

```diff
@@ -0,0 +1,304 @@
+import {Component, For, Index, Show, createEffect, createMemo, on} from "solid-js";
+import {
+  Pagination,
+  SortMarker,
+  TableColumnVisibilityController,
+  TableContextProvider,
+  TableSearch,
+  TableSummary,
+  getHeaders,
+  tableStyle as ts,
+} from ".";
+import {ColumnFilterController, Spinner} from "..";
+
+/** Type of tquery-related information in column meta. */
+export interface TQueryColumnMeta {
+  type: ColumnType;
+}
+
+declare module "@tanstack/table-core" {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  interface ColumnMeta<TData extends RowData, TValue> {
+    tquery?: TQueryColumnMeta;
+  }
+}
+
+export interface ColumnOptions {
+  columnDef?: Partial<IdentifiedColumnDef<object>>;
+}
+
+const TableTranslations = new TranslationEntriesInterface(
+  // "No results" text.
+  "empty_table_text",
+  // Summary of the table, taking the number of rows as count.
+  "summary",
+);
+
+export interface TQueryTableProps {
+  /** The entity URL, must not change. */
+  staticEntityURL: string;
+  /**
+   * The filter that is always applied to the data, regardless of other filtering.
+   * This is used to create e.g. a table of entities A on the details page of a particular
+   * entity B, so only entities A related direclty to that particular entity B should be shown.
+   */
+  intrinsicFilter?: Filter;
+  /**
+   * Prefix of the translation keys used in the table.
+   * If not specified, a generic set of texts is used, e.g. "5 results". By specifying this
+   * prefix, you can customise the table to show e.g. "5 users".
+   */
+  translations?: TranslationEntriesPrefix<typeof TableTranslations>;
+  /** Additional column names. Their options are taken from `columnOptions`. */
+  additionalColumns?: string[];
+  /** Overrides for the definition of specific columns. */
+  columnOptions?: Partial<Record<string, ColumnOptions>>;
+  initialVisibleColumns?: string[];
+  initialPageSize?: number;
+}
+
+export const TQueryTable: Component<TQueryTableProps> = (props) => {
+  const entityURL = props.staticEntityURL;
+  const t = useLangFunc();
+  // eslint-disable-next-line solid/reactivity
+  const tt = TableTranslations.forPrefix(() => props.translations || "tables.tables.generic");
+
+  const SORTABLE_COLUMN_TYPES = new Set<ColumnType>(["string", "decimal0", "decimal2", "bool", "date", "datetime"]);
+
+  type CellTemplate = ColumnDefTemplate<CellContext<object, unknown>>;
+
+  function cellFunc<V>(
+    func: (v: V) => CellTemplate | undefined,
+  ): (c: CellContext<object, unknown>) => CellTemplate | undefined {
+    return (c) => {
+      const val = c.getValue();
+      if (val === undefined) {
+        return undefined;
+      }
+      return func(val as V);
+    };
+  }
+
+  const COLUMN_CELL_BY_TYPE = new Map<ColumnType, CellTemplate>([
+    ["decimal0", cellFunc<number>((v) => String(v))],
+    ["decimal2", cellFunc<number>((v) => v.toFixed(2))],
```

Comment:
pewnie w przyszłości lepiej będzie skorzystać z https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat

żeby np. w języku polskim był przecinek

Reply:
o to już generalnie były spory, chyba Ania mówi, że chce kropkę, i generalnie poprawnego formatowania "1234" "12 234" też nie chcą

ja bym powiedział, że system w języku polskim powinien używać intl'a dla języka polskiego, ale no nie będę tego forsował :D

Reply:
Done.

Reply:
@mblajek tak, tak było xD